benches/bench.rs

```diff
@@ -2,110 +2,149 @@
-            });
+                    black_box(g);
+                },
+                BatchSize::SmallInput,
+            );
         });
     }
+
     group.finish();
 }
 
+/// put: write a fixed payload for N keys into a fresh graph per iteration.
+/// Payload is prepared once in setup; throughput is `Elements(N)`.
 fn bench_put(c: &mut Criterion) {
-    let sizes = [10, 100, 1000, 10_000];
+    let sizes: [usize; 4] = [10, 100, 1000, 10_000];
     let mut group = c.benchmark_group("put");
-    for &size in &sizes {
-        group.bench_with_input(BenchmarkId::from_parameter(size), &size, |b, &size| {
-            let mut graph = setup_graph(size);
-            b.iter(|| {
-                for i in 0..size {
-                    black_box(
-                        graph.put(black_box(i), black_box(&Hex::from_str_bytes("some string"))),
-                    );
-                }
-                black_box(&mut graph);
-            });
+
+    for &n in &sizes {
+        group.throughput(Throughput::Elements(n as u64));
+
+        group.bench_with_input(BenchmarkId::from_parameter(n), &n, |b, &n| {
+            b.iter_batched(
+                || {
+                    let g = setup_graph(n);
+                    let payload = Hex::from_str_bytes("some string");
+                    (g, payload)
+                }, // setup (not measured)
+                |(mut g, payload)| {
+                    // hot path (measured)
+                    for i in 0..n {
+                        black_box(g.put(black_box(i), black_box(&payload)));
+                    }
+                    black_box(g);
+                },
+                BatchSize::SmallInput,
+            );
         });
     }
+
     group.finish();
 }
 
+/// put_and_data: for each key, put payload and then fetch data.
+/// Throughput is `Elements(N)`. Only the hot path is measured.
 fn bench_put_and_data(c: &mut Criterion) {
-    let sizes = [10, 100, 1000, 10_000];
+    let sizes: [usize; 4] = [10, 100, 1000, 10_000];
     let mut group = c.benchmark_group("put_and_data");
-    for &size in &sizes {
-        group.bench_with_input(BenchmarkId::from_parameter(size), &size, |b, &size| {
-            let mut graph = setup_graph(size);
-            b.iter(|| {
-                for i in 0..size {
-                    black_box(
-                        graph.put(black_box(i), black_box(&Hex::from_str_bytes("some string"))),
-                    );
-                    _ = black_box(graph.data(black_box(i)));
-                }
-                black_box(&mut graph);
-            });
+
+    for &n in &sizes {
+        group.throughput(Throughput::Elements(n as u64));
+
+        group.bench_with_input(BenchmarkId::from_parameter(n), &n, |b, &n| {
+            b.iter_batched(
+                || {
+                    let g = setup_graph(n);
+                    let payload = Hex::from_str_bytes("some string");
+                    (g, payload)
+                }, // setup (not measured)
```

Comment:
@RAprogramm we try to avoid inline comments and also empty lines inside methods. They only make code less readable. If you want to explain the code, write a docblock in front of the function.